firestore/integration_test_internal/src/android/env_android_test.cc

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "app/src/util_android.h"
+#include "firestore/src/android/firestore_android.h"
+#include "firestore/src/jni/arena_ref.h"
+#include "firestore/src/jni/env.h"
+#include "firestore/src/jni/hash_map.h"
+#include "firestore/src/jni/long.h"
+
+#include "firestore_integration_test_android.h"
+
+#include "gtest/gtest.h"
+
+namespace firebase {
+namespace firestore {
+namespace jni {
+namespace {
+
+Method<Object> kGet("get", "(Ljava/lang/Object;)Ljava/lang/Object;");
+Method<Object> kPut("put",
+                    "(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;");
+
+class EnvTestAndroid : public FirestoreAndroidIntegrationTest {
+ public:
+  void SetUp() override {
+    FirestoreAndroidIntegrationTest::SetUp();
+    jclass g_clazz = util::map::GetClass();
+    loader().LoadFromExistingClass("java/util/HashMap", g_clazz, kGet, kPut);
+    ASSERT_TRUE(loader().ok());
+  }
+};
+
+TEST_F(EnvTestAndroid, EnvCallTakeArenaRefTest) {
```

Comment:
I try this test but it doesn't work, since the code crash before return to the cpp side:
```
TEST_F(EnvTest, CallsNullArenaRef) {
  ArenaRef null_arena_ref;

  Local<Class> clazz = env().FindClass("java/lang/String");
  jmethodID to_lower_case =
      env().GetMethodId(clazz, "toLowerCase", "()Ljava/lang/String;");

  try {
    Local<Object> result = env().Call(null_arena_ref, Method<Object>(to_lower_case));
    FAIL() << "A NullPointerException should have been thrown.";
  } catch (const std::exception& e) {
    Local<Throwable> exception = env().ClearExceptionOccurred();
    ASSERT_EQ(env().GetObjectClass(exception).GetName(env()), "java.lang.NullPointerException");
    SUCCEED() << "Caught exception";
  }

}
```
The error message is 
`'JNI DETECTED ERROR IN APPLICATION: can't call java.lang.String java.lang.String.toLowerCase() on null object in call to CallObjectMethodV'`

Reply:
The test is not correct: `env().Call()` does _not_ throw a c++ exception; rather, it should set a _pending exception_ on the `Env` object. Try this variation of your test code:

```cpp
TEST_F(EnvTest, CallsNullArenaRef) {
  ArenaRef null_arena_ref;
  Local<Class> clazz = env().FindClass("java/lang/String");
  jmethodID to_lower_case = env().GetMethodId(clazz, "toLowerCase", "()Ljava/lang/String;");
  ASSERT_TRUE(env.ok()) // Make sure that there is no pending exception before calling Call()

  env().Call(null_arena_ref, Method<Object>(to_lower_case));
  Local<Throwable> exception = env().ClearExceptionOccurred();
  ASSERT_TRUE(exception) << "A Java NullPointerException should have been thrown by Call()";
  ASSERT_EQ(env().GetObjectClass(exception).GetName(env()), "java.lang.NullPointerException");
}
````

Reply:
The problem with the above code is the test crash at `env().Call` , when the java side crash the test app crash. And I try to catch the crash using cpp `try` and obviously it doesn't work :(

Reply:
Outcome: This test is invalid. Both Local<T> and Global<T> will cause a crash if specified to env.Call() when they are null, so it is fine if ArenaRef does too.